src/Streamer/HttpCsv.php

```diff
@@ -143,12 +143,12 @@ public function getIterator()
         );
 
         $csvReader = \League\Csv\Reader::createFromStream(StreamWrapper::getResource($response->getBody()));
-        if (empty($this->headers)) {
+        if (empty($this->headers))
+        {
             $csvReader->setHeaderOffset(0);
-        } else {
-            foreach ($csvReader->getIterator() as $row) {
-                yield array_combine($this->headers, $row);
-            }
+        }
```

Comment:
You can also directly return `$csvReader->getIterator()` here, instead of doing the conditional again below.